Docs (workflows): classification/object model performance (#284)
encord-team/encord-active: docs/docs/workflows/evaluate-classification-model.md

# Evaluating classification models

**Visualize the performance of your model**

Encord Active enables you to inspect classification performance metrics such as Accuracy, Precision, Recall, and F1 

> scores as well as confusion matrix. Performance metrics can also be reported according to different class combinations.

Reviewer: ```suggestion
scores as well as a confusion matrix. Performance metrics can also be inspected according to different class combinations.
```

> `Prerequisites:` Dataset, Labels, Classification Predictions

Reviewer: Perhaps we should add a reference to how to import model predictions?

#### Steps

1. Navigate to the _Model Quality_ > _Metrics_ tab on the left sidebar.
2. Under the **Classifications** tab, you will see the main performance metrics (accuracy, mean precision, mean recall, 
and mean F1 scores), metric importance graphs, confusion matrix, and class-based precision and recall plot.
3. You can select classes of interest on the upper bar to customize plots.

> 4. Via Confusion Matrix, you can detect which classes are confused with each other (uni-directional or bi-directional).

Reviewer: ```suggestion
4. Via the confusion matrix, you can detect which classes are confused with each other (uni-directional or bi-directional).
```

5. On the **Precision-Recall** plot, you can observe which classes the model has difficulty in and which classes it 
does well.
6. According to insights you get here, you can, e.g., prioritize from which classes you need to collect more data.

#### Example

The following is the model performance result for a Pokemon dataset (classes: *Charmander, Mewtwo, Pikachu, 
Squirtle*)

![metric](../images/workflows/evaluate-classification-model/img_1.png)  

## Finding Important Metrics 

The three important metrics for this dataset are *Image-level Annotation Quality, Red Values*, and *Image Singularity*. 
When 
we look at their correlation, we see that as the Image-Level Annotation Quality increases, model performance increases, 
too. On the other hand, Red Values and Image Singularity have a negative correlation with the model performance.

> When we look at the confusion matrix, you can see that most of the predictions are correct; Meanwhile, we can easily 

Reviewer: ```suggestion
When we look at the confusion matrix, we find that most of the predictions are correct; Meanwhile, we can 
```

> observe that a significant part of the *charmader* images was predicted as *pikachu*, resulting in low recall for the 

Reviewer: ```suggestion
observe that a significant part of the *charmader* images were predicted as *pikachu*, resulting in low recall for the 
```

> *charmander* and low precision for the *pikachu*. So there might be value in investigating these wrongly labeled 

Reviewer: ```suggestion
*charmander* and low precision for the *pikachu* class. So there might be value in investigating these wrongly labeled 
```
